homework-solutions/js/hw-4/task-3.js

```diff
@@ -1,8 +1,26 @@
 /*
-
-(НЕ ОБЯЗАТЕЛЬНО)
-
 Преобразовать Task 2 таким образом, чтобы значение НАПРИМЕР '2' (т.е. ЛЮБАЯ строка в которой лежат ТОЛЬКО ЦИФРЫ) пропускалось, 
   преобразовываясь в number
-
 */
+
+const minAge = 18;
+const maxAge = 60;
+let age;
+
+age = ["10", 17, "18", 19, "59", 60, "61", null];
+
+age.forEach((age) => {
+  if (!(typeof age === 'number' || typeof age === 'string')) throw new Error('Incorrect data type');
+
+  if (typeof age === 'string') parseInt(age);
+
```

Comment:
Будет полезно посмотреть разницу между `isNaN` и `Number.isNaN`
```suggestion
  if (isNaN(age)) throw new Error('Incorrect data type');
  age = +age; // Number(age)
```

`parseInt` не совсем подходит, потому как может бьіть пример значения `34a`, мьі вьітягиваем `34`, но само ведь значение не подходит под наши условия `ЛЮБАЯ строка в которой лежат ТОЛЬКО ЦИФРЫ`